src/lib.rs

```diff
@@ -284,7 +287,7 @@ cfg_if! {
 
         // These are all inline functions on android, so they end up just being entirely
         // missing on that platform.
-        #[cfg(not(target_os = "android"))]
+        #[cfg(not(any(target_os = "android", target_env = "sgx")))]
```

Comment:
Switch this to use `cfg_if` as well. Are these also inline functions on `sgx` ? In any case, it might be worth it to update the comment.